pkg/oc/admin/diagnostics/diagnostics.go

```diff
@@ -82,50 +80,100 @@ const (
 
 		    %[1]s
 
-		If run without flags, it will check for standard config files for
-		client, master, and node, and if found, use them for diagnostics.
-		You may also specify config files explicitly with flags, in which case
-		you will receive an error if they are not found. For example:
+		Available diagnostics vary based on client config and local OpenShift host config.
+		Config files in standard locations for client, master, and node are used, or
+		you may specify config files explicitly with flags. For example:
 
 		    %[1]s --master-config=/etc/origin/master/master-config.yaml
 
-		* If master/node config files are not found and the --host flag is not
-		  present, host diagnostics are skipped.
-		* If the client has cluster-admin access, this access enables cluster
-		  diagnostics to run which regular users cannot.
-		* If a client config file is not found, client and cluster diagnostics
-		  are skipped.
+		* Explicitly specifying a config file raises an error if it is not found.
+		* A client config with cluster-admin access is required for most cluster diagnostics.
+		* Diagnostics that require a config file are skipped if it is not found.
+		* The standard set also skips diagnostics considered too heavyweight.
 
-		Diagnostics may be individually run by passing diagnostic name as arguments.
+		An individual diagnostic may be run as a subcommand which may have flags
+		for specifying options specific to that diagnostic.
+
+		Finally, the "all" subcommand runs all available diagnostics (including heavyweight
+		ones skipped in the standard set) and provides all individual diagnostic flags.
+		`)
+	longDescriptionAll = templates.LongDesc(`
+		This utility helps troubleshoot and diagnose known problems for an OpenShift cluster
+		and/or local host. This subcommand exists to run all available diagnostics:
+
+		    %[1]s
 
-		    %[1]s <DiagnosticName>
+		Available diagnostics vary based on client config and local OpenShift host config.
+		All flags from the base command work similarly here, but all possible flags for
+		individual diagnostics are also available.
+		`)
+	longDescriptionIndividual = templates.LongDesc(`
+		Runs the %s diagnostic.
 
-		The available diagnostic names are: %[2]s.`)
+		%s
+		`)
 )
 
 // NewCmdDiagnostics is the base command for running any diagnostics.
 func NewCmdDiagnostics(name string, fullName string, out io.Writer) *cobra.Command {
-	o := &DiagnosticsOptions{
-		RequestedDiagnostics: []string{},
-		LogOptions:           &log.LoggerOptions{Out: out},
-		ImageTemplate:        variable.NewDefaultImageTemplate(),
-		NetworkOptions:       &NetworkDiagnosticsOptions{},
+	available := availableDiagnostics()
+	o := &DiagnosticsConfig{
+		RequestedDiagnostics:     available.Names().Difference(defaultSkipDiagnostics()),
+		ParameterizedDiagnostics: types.NewParameterizedDiagnosticMap(available...),
+		LogOptions:               &log.LoggerOptions{Out: out},
 	}
 
 	cmd := &cobra.Command{
 		Use:   name,
```

Comment:
I suggest that you lowercase the names so that we have all subcommands as lowercase. For backwards compatibility we can add the current format as aliases with the `Aliases` attribute.

Reply:
Sorry I meant this for each individual command, in `NewCmdDiagnosticsIndividual`.

Reply:
Alright, will do.